legend-engine-xt-persistence-component/legend-engine-xt-persistence-component-logical-plan/src/main/java/org/finos/legend/engine/persistence/components/planner/UnitemporalDeltaPlanner.java

```diff
@@ -63,6 +63,11 @@ class UnitemporalDeltaPlanner extends UnitemporalPlanner
     {
         super(datasets, ingestMode, plannerOptions);
 
+        // Validate if the optimizationFilters are comparable
+        if (ingestMode.optimizationFilters().isPresent())
```

Comment:
... could then switch this to an empty check

Reply:
Not sure if this is what David was referring, but we should use the functional interfaces to do this, like Optional.ifPresent(x -> ...)

Reply:
Sure, Will do that